propagation_text.go

```diff
@@ -90,8 +189,18 @@ func (textMapPropagator) Extract(
 	}
 
 	return SpanContext{
-		TraceID: traceID,
-		SpanID:  spanID,
-		Baggage: decodedBaggage,
+		TraceID:        traceID,
+		LeadingTraceID: leadingTraceID,
+		SpanID:         spanID,
+		Baggage:        decodedBaggage,
+		TraceState:     opaqueTraceState,
 	}, nil
 }
+
+func compileTraceParentRegexp() {
+	traceParentRegexp, _ = regexp.Compile(`^[[:xdigit:]]{2}-([[:xdigit:]]{16})([[:xdigit:]]{16})-([[:xdigit:]]{16})-[[:xdigit:]]{2}$`)
```

Comment:
While the error does get thrown out, the nil value of `traceParentRegexp` does get used.

The goal was to prevent unnecessary panics downstream. I'm also happy to return an error if the regex is nil, or to use `MustCompile` at the initial declaration time. Beyond not panicking in the middle of a running request, I don't really have strong preferences :)